src/coordinator/storage/converter.go

```diff
@@ -177,3 +183,90 @@ func SeriesToPromSamples(series *ts.Series) []*prompb.Sample {
 
 	return samples
 }
+
+const (
+	initRawFetchAllocSize = 32
+	workerPoolSize        = 10
+)
+
+var (
+	workerPool sync.WorkerPool
+)
+
+func init() {
+	// Set up worker pool
+	workerPool = sync.NewWorkerPool(workerPoolSize)
+	workerPool.Init()
+}
+
+type decompressRequest struct {
+	iter      encoding.SeriesIterator
+	namespace ident.ID
+	result    *ts.Series
+}
+
+func (w *decompressRequest) Process(ctx context.Context) error {
+	iter := w.iter
+	ns := w.namespace
+	if ns == nil {
+		ns = iter.Namespace()
+	}
+
+	metric, err := FromM3IdentToMetric(ns, iter.ID(), iter.Tags())
+	if err != nil {
+		return err
+	}
+
+	datapoints := make(ts.Datapoints, 0, initRawFetchAllocSize)
+	for iter.Next() {
+		dp, _, _ := iter.Current()
+		datapoints = append(datapoints, ts.Datapoint{Timestamp: dp.Timestamp, Value: dp.Value})
```

Comment:
Since you're pre-allocating above, is it better to use an index. e.g.

```
for i := 0; iter.Next(); i++ {
	dp, _, _ := iter.Current()
	datapoints[i] = ts.Datapoint{Timestamp: dp.Timestamp, Value: dp.Value}
}
```

Reply:
I need it to be able to expand if initRawFetchAllocSize is too small